ultranest/popstepsampler.py

```diff
@@ -536,8 +536,219 @@ def __next__(
+                                       
+                
+            
+            interval_final=0. 
+            
+            for k in range(self.nsteps):
+                # Defining scale jitter
+                factor_scale=self.scale_jitter_func()
+                # Defining slice direction
+                v = self.generate_direction(allu, region,scale= 1.0)*self.scale*factor_scale
+                # limite of the slice based on the unit cube boundaries
+                tleft,tright= unitcube_line_intersection(allu, v)
+                # Defining bound of the slice
+                Theta_min_worker,Theta_max_worker = self.slice_limit(tleft,tright)
+                Theta_min,Theta_max=Theta_min_worker.copy(),Theta_max_worker.copy()
+                # Index of the workers working concurrently
+                worker=np.arange(0,self.popsize,1,dtype=int)
+                # Status indicating if a points has already find its next position
+                status=np.zeros(self.popsize,dtype=int) # one for success, zero for running
+                
+                # Loop until each points has found its next position or we reached 100 iterations
+                loop_n=0
+                while (status==0).any() and loop_n<100:
+                
+                    # Sampling points on the slices
+                    Theta=Theta_min_worker+(Theta_max_worker-Theta_min_worker)*np.random.uniform(size=(self.popsize,))
+                
+                    points=allu[worker,:]
+                    v_worker=v[worker,:]
+                    proposed_u=points+Theta.reshape((-1,1))*v_worker
+                
+                    proposed_p = transform(proposed_u)
+                    proposed_L = loglike(proposed_p)
+                    nc+=self.popsize
+                    # Updating the pool of points based on the newly sampled points
+                    Theta_min,Theta_max,worker,status,allu,allL,allp,nth=update_vectorised_slice_sampler(\
+                    Theta,Theta_min,Theta_max,proposed_L,proposed_u,proposed_p,worker,status,Lmin\
+                    ,allu,allL,allp,self.popsize)
+                    n_throws+=nth
+                    # Update of the limits of the slices
+                    Theta_min_worker=Theta_min[worker]
+                    Theta_max_worker=Theta_max[worker]
+                    loop_n+=1
+                # Record of the final interval on theta for scale adaptation
+                interval_final+=np.median(Theta_max-Theta_min)
+
+
+            
+            interval_final=interval_final/self.nsteps
+            
+            
+            self.throwed+=n_throws
+            self.ncalls+=nc
+            
+            assert np.array([p!=np.zeros(ndim) for p in allp]).all(), 'some walkers never moved! Double nsteps of PopulationEllipticalSliceSampler.'
+            self.prepared_samples = list(zip(allu, allp, allL))
+
+            
+            # Scale adaptation such that the final interval is
```

Comment:
The meaning of slice_size as explained here should be better explained in the __init__ docstring. Maybe it should be called something like adapt_slice_length_target or adapt_slice_fraction or adapt_target_slice_padding_factor, because scale and slice_size are indistinguishable at first glance.